compiler/plugins/target/AMD-AIE/iree-amd-aie/Transforms/AMDAIEDistributeCoresAndObjectFifos.cpp

```diff
@@ -491,20 +518,71 @@ LogicalResult insertLogicalObjectFifoAccess(ModuleOp moduleOp) {
             auto accessOp = rewriter.create<AMDAIE::LogicalObjectFifoAccessOp>(
                 rewriter.getUnknownLoc(), std::get<0>(value),
                 std::get<1>(value));
-            linalgOp->setOperand(idx, accessOp);
-          } else if (auto type =
-                         llvm::dyn_cast<MemRefType>(operand.get().getType())) {
-            Value memref = operand.get();
+            op->setOperand(idx, accessOp);
+          } else if (isa<memref::AllocOp, memref::SubViewOp,
+                         vector::TransferReadOp, arith::ExtFOp>(
+                         operand.get().getDefiningOp())) {
+            OpOperand *opOperand = &operand;
```

Comment:
What's needed for these vector ops that's different than for the linalg? I.e. can we not just walk all the vector read, write and contraction ops and perform the same operation as for the linalg, i.e. looping through the operands and inserting acces operations?

Reply:
You're correct. Very good point! I've adapted the code to do that and it looks concise/clean. Thanks!

Based on this I've marked other comments as resolved - since those snippets aren't needed anymore.